office-add-in/src/taskpane/components/OutlookToZaak/steps/StepMetadata.tsx

```diff
@@ -0,0 +1,286 @@
+      form.setValue("selectedItems", nextSelectedItems, {
+        shouldDirty: false,
+        shouldValidate: true,
+      });
+    }
+  }, [fileIds, fileById, form]);
+
+  const handleValidityChange = React.useCallback(
+    (id: string, isValid: boolean) => {
+      const current = form.getValues("validById") || {};
+      if (current[id] === isValid) return;
+      form.setValue(`validById.${id}` as const, isValid, {
+        shouldDirty: false,
+        shouldValidate: false,
+      });
+    },
+    [form]
+  );
+
+  const allValid = selectedItems.length > 0 && selectedItems.every((fileId) => !!validById[fileId]);
+
+  const submit = form.handleSubmit((values) => {
+    const files = (values.selectedItems || [])
+      .map((fileId) => values.filesById[fileId])
+      .filter((row): row is DocumentRow => Boolean(row));
+
+    const payload: SubmitPayload = { files };
+    onSubmit(payload);
+  });
+
+  return (
+    <>
+      <section className={common.title}>
+        <Subtitle1>Bestandsgegevens</Subtitle1>
+        <Body1>Vul bij elk bestand de bijbehorende metadata in.</Body1>
+      </section>
+      <FormProvider {...form}>
+        <form onSubmit={submit}>
+          <Accordion
+            multiple={false}
+            collapsible
+            openItems={openItemPanel === null ? [] : [openItemPanel]}
+            onToggle={(_, { openItems }) => onAccordionToggle(openItems as OpenItems)}
+            className={accordionStyles.accordion}
+          >
+            {selectedItems.map((fileId) => {
+              const displayName = getDisplayName(fileId);
+              return (
+                <AccordionItem key={fileId} value={fileId} className={accordionStyles.item}>
+                  <AccordionHeader className={accordionStyles.header} expandIconPosition="end">
+                    <span>{displayName}</span>
+                    {validById[fileId] && (
+                      <span
+                        aria-label="volledig"
+                        title="Formulier compleet"
+                        style={{ marginLeft: 8, color: "green" }}
+                      >
+                        ✓
+                      </span>
+                    )}
+                  </AccordionHeader>
+                  <AccordionPanel className={accordionStyles.panel}>
+                    <DocumentMetadataPanel
+                      key={fileId}
+                      fileId={fileId}
+                      onValidityChange={handleValidityChange}
+                    />
+                  </AccordionPanel>
+                </AccordionItem>
+              );
+            })}
+          </Accordion>
+
+          <div style={{ display: "flex", gap: tokens.spacingHorizontalM }}>
+            <Button appearance="secondary" type="button" onClick={onBack}>
+              Vorige stap
+            </Button>
+            <Button appearance="primary" type="submit" disabled={!allValid}>
+              Bestanden koppelen
+            </Button>
```

Comment:
Form should be in control of this 

```suggestion
            <Button appearance="primary" type="submit" disabled={!form.formState.isValid}}>
              Bestanden koppelen
            </Button>
```